src/components/PostDetails/PostDetails.jsx

```diff
@@ -1,45 +1,76 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { getPostComments, deletePostComment } from '../../api/comments';
 import { NewCommentForm } from '../NewCommentForm';
 import './PostDetails.scss';
 
-export const PostDetails = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
-
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
-
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
-
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
-
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>sad sds dfsadf asdf asdf</p>
-        </li>
-      </ul>
-    </section>
-
-    <section>
-      <div className="PostDetails__form-wrapper">
-        <NewCommentForm />
-      </div>
-    </section>
-  </div>
-);
+export const PostDetails = ({ posts, selectedPostId }) => {
+  const [comments, setComments] = useState([]);
+  const [hidden, setHidden] = useState(true);
+  const selectedPost = posts.find(post => post.id === selectedPostId);
+
+  useEffect(() => {
+    getPostComments(selectedPostId)
+      .then(setComments);
+  }, [selectedPostId]);
+
+  return (
+    <div className="PostDetails">
+      <h2>Post details:</h2>
+
+      <section className="PostDetails__post">
+        {selectedPost ? <p>{selectedPost.title}</p> : null}
+      </section>
+
+      <section className="PostDetails__comments">
+        <button
+          type="button"
+          className="button"
+          onClick={() => setHidden(!hidden)}
+        >
+          {hidden ? 'Show' : 'Hide'}
+          {' '}
+          {comments.length}
+          {' '}
+          comments
+        </button>
+
+        <ul hidden={hidden} className="PostDetails__list">
+          {comments && comments.map(comment => (
```

Comment:
```suggestion
          {comments && comments.map(comment => (
```
since default comments value is an empty array, this && always will go to the comments.map

Reply:
+